### ISSUE TITLE
Possible non-OSS licensed vstest binary in the VMR

### ISSUE THREAD

**Issue by ellahathaway** [dotnet/source-build#4255]
Upon updating scancode to 32.1.0, a vstest binary was detected as being under a non-OSS license:

```json  
  {
      "path": "test/Microsoft.TestPlatform.CoreUtilities.UnitTests/TestAssets/dotnetWinX86.exe",
      "detected_license_expression": "proprietary-license"
    }
```

https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/rules/proprietary-license_975.RULE

This binary should be removed from the VMR. The allowed-vmr-binaries should also potentially be updated.

**Comment by mthalman** [comment]
No idea why it think it's a proprietary license. It's only meant to scan text files so who knows what it's finding in a binary file. But yeah, all of the files in this directory are binary: https://github.com/dotnet/dotnet/tree/main/src/vstest/test/Microsoft.TestPlatform.CoreUtilities.UnitTests/TestAssets. Note too the files with no file extension are also binaries. Are those being detected by the binary scanner?

**Comment by ellahathaway** [comment]
> Are those being detected by the binary scanner?

Yes, the binaries in that directory are detected by the binary scanner, but they are currently allowed in the VMR under https://github.com/dotnet/dotnet/blob/f7296fb63b584e0e7a5cac463daac7183c9d8d9e/eng/allowed-vmr-binaries.txt#L124-L125.

> It's only meant to scan text files

Since we're now allowing binaries into the VMR, should we scan binary files for non-OSS licenses? I was under the impression that the scanner scanned binary and text files. 

The permissible sources documentation describes that non-OSS licensed binaries are not allowed in the VMR: https://github.com/dotnet/dotnet/blob/f7296fb63b584e0e7a5cac463daac7183c9d8d9e/src/arcade/Documentation/UnifiedBuild/VMR-Permissible-Sources.md?plain=1#L85

**Comment by mthalman** [comment]
> Looking at the scanner, I see that [certain binary extensions are not being excluded](https://github.com/dotnet/dotnet/blob/main/test/Microsoft.DotNet.SourceBuild.SmokeTests/LicenseScanTests.cs#L106-L133) (exe, dll, etc).

That's because at the time this was authored, there were no such binary files in the VMR.

> Since we're now allowing binaries into the VMR, should we scan all binary files for non-OSS licenses? I was under the impression that the license scanner scans binary and text files.

While the tool does say it supports scanning binary files, I can't see how that provides any accurate information regarding the license of such a file. In other words, a binary file can potentially have a license without containing any mention of that license in the file itself. I'm guessing they're just running the equivalent of [`strings`](https://www.man7.org/linux/man-pages/man1/strings.1.html) on the file and scanning that content. That being said, we may as well scan them as long as they're scannable.
